Implement vertical jumping & fix offsets for jumps right before/after caret

jannes opened the pull request:
Hey, 

I added the vertical jumps without bounds just like the vim-sneak plugin behavior.
Also I noticed that with the offsets you used, jumps directly after the current character 
are not allowed, e.g in "cha" if the cursor is on "c" and you search "ha" it didn't jump there.
But the vim-sneak plugin would jump in that case.

Hope you agree with the changes:)

Edit: The original vim-sneak plugin does not ignore case, so I also removed that.
I guess you like to have that behavior, but I would argue to replicate the vim-sneak behavior exactly,
if you call this a port of it.

jannes commented:
Just saw some of your tests are failing.
I'll have a look at that later and also add some more tests

Mishkun commented:
@jannes Thanks for the PR! Can you pull master branch into this so we can get checks up and running?

jannes commented:
I had already rebased on master. I actually do not know how CI works on Github,
but I see it ran on my fork repo (and passed).

I just changed two tests and added 4 more (see the description from the commit).
I am not absolutely sure if that caret position from the one test is impossible, but
in my understanding when placed over the last character in a file it should be left of that character.

Mishkun reviewed (approved):
Checked this out, everything looks fine, tests are passing. You removed case insensitivity, and with vertical jumping I thing it is a right thing to do so.
